src/js/components/widgets-link/widget-link-number.react.js

```diff
@@ -0,0 +1,287 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Modal from 'react-bootstrap/lib/Modal';
+import Button from 'react-bootstrap/lib/Button';
+import DraggableModal from '../draggable-modal.react';
+import Icon from '../icon.react';
+
+import { editor } from '../../editor/editor';
+
+/**
+ * Represents a widget link for a number
+ * Gets created on click, as opposed to normal widgets that get created on editor parse
+ */
+export default class WidgetLinkNumber extends React.Component {
+    /**
+     * Used to setup the state of the component. Regular ES6 classes do not
+     * automatically bind 'this' to the instance, therefore this is the best
+     * place to bind event handlers
+     *
+     * @param props - parameters passed from the parent
+     */
+    constructor (props) {
+        super(props);
+        this.state = {
+            displayPicker: this.props.display
+        };
+
+        this.cursor = this.props.cursor;
+        this.match = this.props.match;
+
+        let linePos = { line: this.cursor.line, ch: this.match.start }; // Position where user cliked on a line
+        this.x = editor.charCoords(linePos).left;
+        this.y = editor.charCoords(linePos).bottom - 80;
+
+        this.fnColor = 'rgb(230, 230, 230)';
+        this.selColor = 'rgb(40, 168, 107)';
+        this.dimColor = 'rgb(100, 100, 100)';
+
+        this.width = 250;
+        this.height = 40;
+
+        this.min = -1;
+        this.max = 1;
+        this.range = this.max - this.min;
+        this.overPoint = false;
+        this.prevOffset = 0;
+        this.prevWheelOffset = 0;
+        this.scale = 2;
+        this.center = (this.width / this.scale);
+        this.offsetX = this.value * this.center;
+
+        this.value = parseFloat(this.props.value);
+        this._setValue(this.value);
+
+        this.drag = false;
+
+        this.handleClick = this.handleClick.bind(this);
+        this._setValue = this._setValue.bind(this);
+        this._drawCanvas = this._drawCanvas.bind(this);
+        this.onMouseDown = this.onMouseDown.bind(this);
+        this.onMouseMove = this.onMouseMove.bind(this);
+        this.onMouseUp = this.onMouseUp.bind(this);
+        this.onMouseLeave = this.onMouseLeave.bind(this);
+        this.onWheel = this.onWheel.bind(this);
+    }
+
+    /**
+     * React lifecycle method called once DIV is mounted
+     */
+    componentDidMount () {
+        this._drawCanvas();
+    }
+
+    /**
+     * Draws the canvas
+     */
+    _drawCanvas () {
```

Comment:
Let's also follow AirBnb style guide: ["Do not use underscore prefix for internal methods of a React component."](https://github.com/airbnb/javascript/tree/master/react#methods)